cmd/skaffold/app/cmd/flags.go

```diff
@@ -657,6 +657,31 @@ var flagRegistry = []Flag{
 		FlagAddMethod: "BoolVar",
 		DefinedOn:     []string{"dev", "render", "run", "debug", "deploy"},
 	},
+	{
+		Name:     "disable-multi-platform-build",
+		Usage:    "When set to true, forces only single platform image builds even when multiple target platforms are specified. Enabled by default for `dev` and `debug` modes, to keep dev-loop fast",
+		Value:    &opts.DisableMultiPlatformBuild,
+		DefValue: false,
+		DefValuePerCommand: map[string]interface{}{
+			"dev":   true,
+			"debug": true,
+		},
+		FlagAddMethod: "BoolVar",
+		DefinedOn:     []string{"build", "dev", "run", "debug"},
+	},
+	{
+		Name:     "check-cluster-node-platforms",
+		Usage:    "When set to true, images are built for the target platforms matching the active kubernetes cluster node platforms. Enabled by default for `dev`, `debug` and `run`",
+		Value:    &opts.CheckClusterNodePlatforms,
+		DefValue: false,
+		DefValuePerCommand: map[string]interface{}{
+			"dev":   true,
+			"debug": true,
+			"run":   true,
+		},
+		FlagAddMethod: "BoolVar",
+		DefinedOn:     []string{"build", "dev", "run", "debug"},
```

Comment:
same here. Does this need to happen on build?

Reply:
https://github.com/GoogleContainerTools/skaffold/pull/7769#discussion_r945112872